### ISSUE TITLE
Allow expressions for my_constants

### ISSUE THREAD

**Issue by dpgrote** [ECP-WarpX/WarpX#1868]
Would it be possible to allow an expression to be used for a my_constants that uses predefined quantities? An example use would be `my_constants.mi = 0.1*m_p`.

**Comment by WeiqunZhang** [comment]
I think this is already supported in `makeParser` function in WarpXUtil.cpp.  The predefined constants there include `q_e`, `m_e`, `m_p`, `epsilon0`, `clight` and `pi`.

**Comment by dpgrote** [comment]
There seems to be a bug somewhere. If I have only the one line `my_constants.nn = 0.1*m_p`, I get this error

ParmParse::queryval type mismatch on value number 0 of 
 last occurence of my_constants.nn
 Expected an "d" type  which can't be parsed from the string "0.1*m_p"
my_constants.nn(nvals = 1)  :: [0.1*m_p]
amrex::Abort::SIGABRT

But if I define another constant, like this:
`my_constants.nn = 1.e1
my_constants.mi = 0.1*m_p
`
It works.

**Comment by WeiqunZhang** [comment]
What if you put `0.1*m_p` inside a pair of `"`?

**Comment by dpgrote** [comment]
Nope, same error

**Comment by WeiqunZhang** [comment]
Maybe the problem is it's not using `queryWithParser`?

**Comment by dpgrote** [comment]
Yes, thanks, that seems to work. I'll post a PR. Though it risks an infinite loop `my_constants.x = x`.

In the case above that was not returning an error, it was because the constant (mi) was not being used in the input file so was not read in.

**Comment by ax3l** [comment]
Ha, we am currently using expressions such as:
```
my_constants.theta = 5. * pi / 180.
my_constants.r0    = 2.0e-6 * cos(5. * pi / 180.)
```
and they are just silently wrong :sweat_smile:

**Comment by dpgrote** [comment]
This will be resolved by PR #1874.
